README.md

```diff
@@ -210,53 +210,72 @@ You will additionally define flags and handle configuration in your init() funct
 For example cmd/root.go:
 
 ```go
+package cmd
+
 import (
-  "fmt"
-  "os"
+	"fmt"
 
-  homedir "github.com/mitchellh/go-homedir"
-  "github.com/spf13/cobra"
-  "github.com/spf13/viper"
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 )
 
-var cfgFile string
+var (
+	// Used for flags.
+	cfgFile     string
+	userLicense string
+
+	rootCmd = &cobra.Command{
+		Use:   "cobra",
+		Short: "A generator for Cobra based Applications",
+		Long: `Cobra is a CLI library for Go that empowers applications.
+This application is a tool to generate the needed files
+to quickly create a Cobra application.`,
+	}
+)
+
+// Execute executes the root command.
+func Execute() error {
+	return rootCmd.Execute()
+}
 
 func init() {
-  cobra.OnInitialize(initConfig)
-  rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.cobra.yaml)")
-  rootCmd.PersistentFlags().StringVarP(&projectBase, "projectbase", "b", "", "base project directory eg. github.com/spf13/")
-  rootCmd.PersistentFlags().StringP("author", "a", "YOUR NAME", "Author name for copyright attribution")
-  rootCmd.PersistentFlags().StringVarP(&userLicense, "license", "l", "", "Name of license for the project (can provide `licensetext` in config)")
-  rootCmd.PersistentFlags().Bool("viper", true, "Use Viper for configuration")
-  viper.BindPFlag("author", rootCmd.PersistentFlags().Lookup("author"))
-  viper.BindPFlag("projectbase", rootCmd.PersistentFlags().Lookup("projectbase"))
-  viper.BindPFlag("useViper", rootCmd.PersistentFlags().Lookup("viper"))
-  viper.SetDefault("author", "NAME HERE <EMAIL ADDRESS>")
-  viper.SetDefault("license", "apache")
+	cobra.OnInitialize(initConfig)
+
+	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.cobra.yaml)")
+	rootCmd.PersistentFlags().StringP("author", "a", "YOUR NAME", "author name for copyright attribution")
+	rootCmd.PersistentFlags().StringVarP(&userLicense, "license", "l", "", "name of license for the project")
```

Comment:
I not convinced about providing such example flags. This is extracted from a very specific tool/app, which is the cobra generator/tool. The fact that both the library and the cmd tool have the same name and are located in the same repo might already be confusing for new users. Providing an example that is based on this so special *inception* might lead users to think that flags `author` or `userLicense` have any effect at all on the license of the app they are building, which is not the case.

If we want to copy some "real world" example, I think we'd better pick any other (i.e. hugo). Furthermore, I think we should have a proper "examples" or "guide" subdir with MWEs. A cobra project can be described in a single markdown file. Ref #910.

Of course, this is not something you introduced. Hence, I completely agree if you consider it to be out of scope of this PR.

Reply:
This PR is just to update the documentation that already exists.  It could be useful to have some sort of more in-depth documentation, however, a list of projects that Cobra is used in is already in the readme so examples such as Hugo are only a few clicks away.

Reply:
So, don't you think that all the flags and viper binds in this `init` function can be misleading rather than helpful? I mean to just remove lines 246-252 from the current proposal.

Reply:
I honestly don't believe this example to be all that confusing.  To be clear, this does not add _more_ flags to the init function example, but rather just updates it with a more current reflection of Cobra's own `cmd/root.go'.

I'm not saying there aren't changes to be made to improve documentation.  I'm saying that because Cobra is widely used, documentation changes need to be well thought through.